src/inference/src/dev/threading/cpu_streams_executor.cpp

```diff
@@ -312,12 +315,71 @@ struct CPUStreamsExecutor::Impl {
         std::vector<int> _cpu_ids;
 #endif
     };
+    // if the thread is created by CPUStreamsExecutor, the Impl::Stream of the thread is stored by tbb Class
+    // enumerable_thread_specific, the alias is ThreadLocal, the limitations of ThreadLocal please refer to
+    // https://spec.oneapi.io/versions/latest/elements/oneTBB/source/thread_local_storage/enumerable_thread_specific_cls.html
+    // if the thread is created by customer, the Impl::Stream of the thread will be stored in variable _stream_map, and
+    // will be counted by thread_local t_stream_count_map.
+    // when the customer's thread is destoryed, the stream's count will became 1,
+    // Call local() will reuse one of them, and release others.
+    class CustomThreadLocal : public ThreadLocal<std::shared_ptr<Stream>> {
```

Comment:
As for me it is bad naming `Custom` ... It would be better `OpenVINOCPUThreadLocal` or `OpenVINOThreadLocal`, because `Custom` is not self-explanatory

Reply:
@redradist  thanks very much for your suggestion, from my point of view, this PR is a work around to fix this issue, the Class CustomThreadLocal is in ov::thread::CPUStreamsExecutor::Impl, the Class CustomThreadLocal is not a dev API which can be used by plugins, and it's not also a API which can be used by users